client/server/pages/index.js

```diff
@@ -293,7 +294,13 @@ function setUpLoggedInRoute( req, res, next ) {
 				// Setting user in the state is safe as long as we don't cache it
 				req.context.store.dispatch( setCurrentUser( data ) );
 
-				if ( data.localeSlug ) {
+				if (
+					data.localeSlug &&
+					! (
+						data.use_fallback_for_incomplete_languages &&
+						isTranslatedIncompletely( data.localeVariant || data.localeSlug )
```

Comment:
Yes, `de` vs `de_formal` is the perfect example, also because one of them is translated completely and the other isn't. The check returns wrong result if passed only `localeSlug`.

There are more places where this kind check is probably wrong.